tests/filters/test_wordwrap.py

```diff
@@ -0,0 +1,283 @@
+
+
+def test_wordwrap_with_autoescape_off(assert_render):
+    template = (
+        "{% autoescape off %}{{ a|wordwrap:3 }} {{ b|wordwrap:3 }}{% endautoescape %}"
+    )
+    assert_render(
+        template,
+        {"a": "a & b", "b": "a & b"},
+        "a &\nb a &\nb",
+    )
+
+
+def test_wordwrap_very_long_text(template_engine):
+    long_text = (
+        "this is a long paragraph of text that really needs"
+        " to be wrapped I'm afraid " * 20_000
+    )
+    template = template_engine.from_string("{{ text|wordwrap:10 }}")
+    result = template.render({"text": long_text})
+
+    # Verify the result contains the expected wrapped pattern
+    expected_pattern = (
+        "this is a\nlong\nparagraph\nof text\nthat\nreally\nneeds to\nbe wrapped"
+    )
+    assert expected_pattern in result
+
+
+def test_wordwrap_with_safe(assert_render):
+    template = "{{ a|wordwrap:3 }} {{ b|safe|wordwrap:3 }}"
+    assert_render(
+        template,
+        {"a": "a & b", "b": "a & b"},
+        "a &amp;\nb a &\nb",
+    )
+
+
+def test_wordwrap_no_argument(assert_parse_error):
+    assert_parse_error(
+        template="{{ text|wordwrap }}",
+        django_message="wordwrap requires 2 arguments, 1 provided",
+        rusty_message="""\
+  × Expected an argument
+   ╭────
+ 1 │ {{ text|wordwrap }}
+   ·         ────┬───
+   ·             ╰── here
+   ╰────
+""",
+    )
+
+
+def test_wordwrap_invalid_width_string(assert_render_error):
+    assert_render_error(
+        template="{{ text|wordwrap:'invalid' }}",
+        context={"text": "hello"},
+        exception=ValueError,
+        django_message="invalid literal for int() with base 10: 'invalid'",
+        rusty_message="""\
+  × Couldn't convert argument ('invalid') to integer
+   ╭────
+ 1 │ {{ text|wordwrap:'invalid' }}
+   ·                  ────┬────
+   ·                      ╰── argument
+   ╰────
+""",
+    )
+
+
+def test_wordwrap_negative_width(assert_render_error):
+    assert_render_error(
+        template="{{ text|wordwrap:-5 }}",
+        context={"text": "hello world"},
+        exception=ValueError,
+        django_message="invalid width -5 (must be > 0)",
+        rusty_message="invalid width -5 (must be > 0)",
+    )
+
+
+def test_wordwrap_bool_zero_width(assert_render_error):
+    assert_render_error(
+        template="{{ text|wordwrap:False }}",
+        context={"text": "hello world"},
+        exception=ValueError,
+        django_message="invalid width 0 (must be > 0)",
+        rusty_message="invalid width 0 (must be > 0)",
+    )
+
+
+def test_wordwrap_zero_width(assert_render_error):
+    assert_render_error(
+        template="{{ text|wordwrap:0 }}",
+        context={"text": "hello world"},
+        exception=ValueError,
+        django_message="invalid width 0 (must be > 0)",
+        rusty_message="invalid width 0 (must be > 0)",
+    )
+
+
+@pytest.mark.skip("Should overflow in django")
```

Comment:
Can you elaborate on why this needs to be skipped?

Reply:
> I think this is another of those situations where it makes sense to seek a fix upstream in Django rather than diverge in API here.

Wouldn't it better to fix the behavior here to align with upstream ? ie ensure we don't Overflow and just fallback to +inf if we fail to convert here ?